tests/params/test_tensor.py

```diff
@@ -0,0 +1,243 @@
+import numpy as np
+import pytest
+
+import porepy as pp
+
+
+def fourth_order_tensor_for_testing(
+    num_cells: np.array, return_tensor_and_matrix: bool
+):
+    """Construct a fourth order tensor with custom fields for testing.
+
+    The tensor which is constructed here is for an anisotropic and homogeneous medium.
+
+    Parameters:
+        return_tensor_and_matrix: Flag telling whether only the tensor should be
+            returned (False) or both the tensor and the matrix representation of the
+            tensor for one cell (True).
+
+    Returns:
+        The fourth order tensor with custom fields, or the tensor and the matrix
+        representation of the tensor for one cell.
+
+    """
+    # The various matrices serve as the foundation for building a tensor for specific
+    # parameters. The cell-wise parameter value for each basis is determined by a 1D
+    # array, referred to as a "field". The matrices that form the basis for different
+    # parameters, along with their corresponding fields, are organized as tuples. These
+    # tuples are later provided to FourthOrderTensor when constructing the tensor.
+    matrix_and_field_1 = (
```

Comment:
Any reason we need five of these? I would think that one or two would suffice.

Reply:
Also, since someone will wonder about this in a few years time: Is there any rational behind the patterns in these matrices, or are they essentially meaningless?

Reply:
I wanted to keep the tensor meaningful, and the easiest choice I had for that was those five matrices (as I use them in my own work and already have them easily accessible). They correspond to a transversely isotropic media, so yes, there is a rationale behind them.  I can adapt it to fewer matrices, but then the tensor might be meaningless.

Reply:
I suggest to make a comment of the physical interpretation and leave it at that.